src/pages/signup/Signup.tsx

```diff
@@ -1,44 +1,72 @@
-import React, { FC, ReactElement, useEffect, useState } from 'react';
+import React, { ChangeEvent, FC, FormEvent, ReactElement, useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 
-import { Button, Input } from 'components';
-import { fakeOnChange } from 'utils';
 import { signup } from 'api/authApi';
-import { DEFAULT_USER_STATE, GAME_NAME, LINK_TEXTS, MOCK_USER_STATE, PAGE_NAMES } from 'constants/commonConstants';
+import { Button, Input } from 'components';
+import {
+  checkEmail,
+  checkFieldNotEmpty,
+  checkPassword,
+  checkPhone,
+  checkButtonDisable,
+} from 'utils';
+import { DEFAULT_USER_STATE, GAME_NAME, PAGE_NAMES, userKeys, userType } from 'constants/commonConstants';
+import { LINK_TEXTS } from 'constants/linkConstants';
 import { ROUTE_CONSTANTS } from 'constants/routeConstants';
 import { BUTTON_TEXTS } from 'constants/buttonConstants';
 import { ERROR_CONSTANTS } from 'constants/errorConstants';
 
 const Signup: FC = (): ReactElement => {
-  const [ state, setState ] = useState(DEFAULT_USER_STATE);
+  const [ state, setState ] = useState<userType>(DEFAULT_USER_STATE);
+  const [ disabled, setDisabled ] = useState<boolean>(true);
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    const { value } = e.target;
+    const name = e.target.name as userKeys;
+    const newState = Object.assign({}, state);
+    newState[name] = value;
+    setState(newState);
+  };
+
+  useEffect(() => {
+    const newDisable = checkButtonDisable();
+    setDisabled(newDisable);
+  }, [state]);
 
-  useEffect(() => setState(MOCK_USER_STATE), []);
+  const signupHandler = async () => {
+    try {
+      await signup(state);
+    } catch (err) {
+      console.error(err?.response?.data?.reason || err?.message || ERROR_CONSTANTS.DEFAULT_ERROR);
+    }
+  };
 
-  const signupHandler = () => {
-    signup(state)
-      .catch(err => console.error(err?.response?.data?.reason || err?.message || ERROR_CONSTANTS.DEFAULT_ERROR));
+  const onSubmitHandler = (e: FormEvent) => {
+    e.preventDefault();
+    signupHandler();
   };
 
   return (
     <div className='main'>
       <div className='content-wrapper double'>
         <h1>{GAME_NAME}</h1>
-        <form className='content'>
+        <form className='content' onSubmit={onSubmitHandler} >
           <h2>{PAGE_NAMES.REGISTRATION}</h2>
           <div className='input-wrapper'>
-            <Input value={state.first_name} name='first_name' title='first name' onChange={fakeOnChange} />
-            <Input value={state.second_name} name='second_name' title='second name' onChange={fakeOnChange} />
+            <Input value={state.first_name} name='first_name' title='first name' onChange={onChange} errorText={checkFieldNotEmpty(state.first_name)} />
```

Comment:
❗️
Длинновые строки получились, давай слегка отформатируем. По самой логике выше писал, что в `errorText` надо передавать не просто функцию, которая "на месте" вычислит ошибку, а хранимую в самом компоненте
```suggestion
            <Input
              value={state.first_name}
              name='first_name'
              title='first name'
              onChange={onChange}
              errorText={checkFieldNotEmpty(state.first_name)}
            />
```